Support Manjaro (Arch Linux Distribution)

PauloPortugal opened the pull request:
### What
[memongo](https://github.com/benweissmann/memongo#caveats-and-notes), [dp-mongo-in-memory](https://github.com/ONSdigital/dp-mongodb-in-memory#supported-versions) and [MongoDB](https://www.mongodb.com/download-center/community/releases) do not support Arch/Manjaro distribution, so this is an attempt to support a [Manjaro](https://manjaro.org/) distribution.

This avoids having to download the MongoDBSpec and assumes the user has a working `mongod` daemon working on the expected `CachedPath`. It makes use of the `XDG_CACHE_HOME` environment variable.


### How to review

Run locally, namely Ubuntu and Mac OS'.

### Who can review

@rafahop & @saminahbab you are probably the best folks to review this :)

PauloPortugal committed:
Support Manjaro (Arch Linux Distribution)

PauloPortugal commented on download/downloadSpec.go:
```diff
@@ -141,6 +141,9 @@ func detectLinuxId() (string, error) {
 	id := osRelease["ID"]
+	if id == "manjaro" {
+		return "manjaro", nil
+	}
```
Arch/Manjaro does not have a `VERSION_ID` property on `/etc/os-release` file, so there is not point in continuing

PauloPortugal replied:
@rafahop I wonder if we should support other Linux versions here... 
```suggestion
	if osName == "linux" & !strings.Contains("debian ubuntu", id) {
		return id, nil
	}
```

rafahop replied:
Other linux versions might be using the os-release file? We'll need to identify the version in most cases to download the appropriate tar file.
When/if there is a need for extending it to other linux versions we could change this appropriately.

rafahop reviewed (changes requested):
I think we are breaking the design with this change by bringing spec code into the config file.
I would suggest adding support for Manjaro and build a "fake" url following the same pattern. It won't find the downloadable file but if we add it manually to the cache it should work. We should document this anyway as this is a special case.

 I believe we could easily achieve that by just adding the manjaro case to the `detectLinuxId` function? It will build a url like `"https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-manjaro-5.0.4.tgz"` and if we manually created the `{CACHE}/mongodb-linux-x86_64-manjaro-5.0.4.tgz` folder and placed `mongod` in there, it wouldn't try and download it and hopefully all works?

PauloPortugal commented:
> I believe we could easily achieve that by just adding the manjaro case to the detectLinuxId function?

@rafahop that was my first option, but the `/etc/os-release` does not have the `VERSION_ID` so   this will fail

```
        versionString := strings.Split(osRelease["VERSION_ID"], ".")[0]
	version, versionErr := strconv.Atoi(versionString)
	if versionErr != nil {
		return "", &UnsupportedSystemError{msg: "invalid version number " + versionString}
	}
```
I would have to create an exception to it.

PauloPortugal committed:
Do not bring the spec code into the config.go

Instead in this scenario we check if the downloadUrl contains a string
called "DO_NOT_DOWNLOAD".

PauloPortugal commented on download/config.go:
```diff
@@ -45,22 +45,33 @@ func NewConfig(ctx context.Context, mongoVersionStr string) (*Config, error) {
+	if downloadUrl == DoNotDownload {
+		basePath, err := defaultBaseCachePath()
+		if err != nil {
+			log.Error(ctx, "cache directory not found", err)
+			return nil, err
+		}
+		return &Config{
+			mongoVersion: *version,
+			cachePath:    basePath + "/mongod",
+		}, nil
+	} else {
+		cachePath, err := buildBinCachePath(ctx, downloadUrl)
+		if err != nil {
+			return nil, err
+		}
 
-	cachePath, err := buildBinCachePath(ctx, downloadUrl)
-	if err != nil {
-		return nil, err
+		return &Config{
+			mongoVersion: *version,
+			mongoUrl:     downloadUrl,
+			cachePath:    cachePath,
+		}, nil
```
@rafahop if this is still not acceptable from you POV, then I will not merge this PR and will keep it rebased with `main`
Thanks for all the patience on this :)

rafahop replied:
Yes, I think keeping this in a separate branch for specific use would be best. It does not really fit the purpose of the library and requires some manual steps for Manjaro so yes, I wouldn't merge it into the main branch.

PauloPortugal replied:
No worries! I completely understand it :)

PauloPortugal commented:
No relevant for the main branch, as only two developer use Arch/Manjaro OS.
This branch will be kept up to date with main.

FYI @saminahbab
